data-serving/data-service/src/model/case.ts

```diff
@@ -1,43 +1,289 @@
 import mongoose from 'mongoose';
+import { ObjectId, Double } from 'mongodb';
 
-export const validOutcomes = ['pending', 'recovered', 'death'];
+export enum Sex {
+    Female,
+    Male,
+    Other,
+}
 
-const demographicsSchema = new mongoose.Schema({
-    age: {
+export enum Outcome {
+    Pending,
+    Recovered,
+    Death,
+}
+
+export enum Species {
+    HomoSapien,
+}
+
+export enum TravelPurpose {
+    Family,
+    Conference,
+    Work,
+}
+
+const dateValidator = {
+    type: Date,
+    min: '2019-11-01',
+    max: Date.now,
+};
+
+const rangeSchema = new mongoose.Schema({
+    minimum: {
+        type: Number,
+        min: -1,
+        max: 200,
+    },
+    maximum: {
         type: Number,
         min: 0,
         max: 200,
     },
+});
+
+const ageSchema = new mongoose.Schema({
+    years: {
+        type: Number,
+        min: -0.75,
+        max: 200,
+    },
+    range: rangeSchema,
+});
+
+const demographicsSchema = new mongoose.Schema({
+    ageSchema,
+    species: {
+        type: String,
+        enum: Object.keys(Species),
+    },
     sex: {
         type: String,
-        enum: ['female', 'male'],
+        enum: Object.keys(Sex),
     },
 });
 
-const caseSchema = new mongoose.Schema({
-    date: {
-        type: Date,
-        min: '2019-11-01',
-        max: Date.now,
-        required: 'Enter a date.',
+const eventSequenceSchema = new mongoose.Schema({
+    onsetSymptoms: dateValidator,
+    confirmed: dateValidator,
+    hospitalAdmission: dateValidator,
+    deathOrDischarge: dateValidator,
+});
+
+const dictionaryValueSchema = new mongoose.Schema({
+    provided: {
+        type: [String],
+        uniqueItems: true,
+    },
+    imputed: {
+        type: [String],
+        uniqueItems: true,
+    },
+    other: {
+        type: [String],
+        uniqueItems: true,
+    },
+});
+
+const metadataSchema = new mongoose.Schema({
+    moderatorId: {
+        type: String,
+        required: 'Enter a moderator id',
```

Comment:
generally, they're fields with >90% completeness in the existing data; in this case (moderatorId), I think it just makes sense that new data coming in should have its origin recorded, since we can control that in our system. Confirmed with the epi folks that this initial set makes sense